### PR TITLE
Fabo/polkadot signer

### PR DESCRIPTION
**Pull request by faboweb**
Draft of tx generation and signing in polkadot

Missing:
- extract fees

Needs: https://github.com/luniehq/lunie-api/pull/461

### CODE REVIEW

File: src/components/common/TmSessionExtension.vue [hunk 1]
```diff
@@ -62,15 +62,6 @@ export default {
...
   },
-  async created() {
-    if (this.$route.params.address && this.$route.params.network) {
-      await this.signIn({
-        address: this.$route.params.address,
-        network: this.$route.params.network
-      })
-      this.$router.push("/portfolio")
-    }
-  },
```
Comment (Bitcoinera) [inline src/components/common/TmSessionExtension.vue]: Shouldn't this be another PR 😛 ?

File: src/routes.js [hunk 2]
```diff
@@ -316,3 +315,27 @@ export default (apollo, store) => {
...
+    fetchPolicy: "cache-first"
+  })
+
+  await store.dispatch(`signIn`, {
+    sessionType: `extension`,
+    address: to.params.address,
+    networkId: to.params.network
+  })
+  next(`/${network.slug}/portfolio`)
+}
```
Comment (Bitcoinera) [inline src/routes.js]: This side change with the routing should be another PR no? ;)

Or why is it needed here?

File: src/components/common/TmSessionExtension.vue (hunk 1 above)
Comment (faboweb) [reply src/components/common/TmSessionExtension.vue]: left overs, I can clean it up if necessary

Comment (Bitcoinera) [reply src/components/common/TmSessionExtension.vue]: Please, split PRs to make more readable & understandable💃

Comment (faboweb) [reply src/components/common/TmSessionExtension.vue]: sure when I clean this up

**Comment by faboweb** [comment]
one strange test error

File: src/ActionModal/utils/networkMessages/polkadot-testnet.js [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
+
+// will only be inited once per session
+let api
+async function getAPI() {
+  if (api) {
+    return api
+  }
+  api = new ApiPromise({
+    provider: new WsProvider(
+      "wss://kusama-rpc.polkastats.io/apikey/HunRG7eUwMTjaBAkz1A6GU1MqBJapaYsYfmU5EVzpAebMr8/"
```
Comment (jbibla) [inline src/ActionModal/utils/networkMessages/polkadot-testnet.js]: APIkey should be `ENV` variable?

Comment (faboweb) [reply src/ActionModal/utils/networkMessages/polkadot-testnet.js]: whole thing should just be in the api. will need to consider rotating the apikey @mariopino

Comment (mariopino) [reply src/ActionModal/utils/networkMessages/polkadot-testnet.js]: No problem as that was only for testing and I'll remove the exposed RPC, now we are using parity RPC.

btw tested and works awesome @faboweb!!
